vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -660,6 +662,7 @@ protected void onDetach(DetachEvent detachEvent) {
             // itself when its parent, for example a dialog, gets attached
             // again.
             setOpened(false);
+            autoAddController.remove();
```

Comment:
Notification has custom logic for auto-removing itself whenever it is detached from the UI. This was added to prevent notifications that have been auto-added to a modal dialog to show up again whenever that dialog opens again. This was done by closing the notification, which now doesn't trigger auto-removal as the closed event from the client-side is not fired.

To keep this working I added a method to `OverlayAutoAddController` to allow the component to auto-remove itself when appropriate.